src/main/java/net/snowflake/client/core/IncidentUtil.java

```diff
@@ -71,6 +68,8 @@ public static String oneLiner(String prefix, Throwable thrown) {
   public static void dumpVmMetrics(String incidentId) {
     PrintWriter writer = null;
     try {
+      File dmpFileDirectory = new File(EventUtil.getDumpPathPrefix());
+      dmpFileDirectory.mkdirs();
```

Comment:
By default, the function EventUtil.getDumpPathPrefix() adds a subdirectory, snowflake_dump, to the end of the dump path. On line 72, we create this subdirectory if it does not yet exist because otherwise the gzip output logic later on in the function cannot figure out where to put the dmp file because of of the directories in the filepath does not exist.